<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>conf_appClass</class>
 <widget class="QMainWindow" name="conf_appClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>514</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Medfisis Configurator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="conf_app.qrc">
    <normaloff>:/conf_app/conf_app.ico</normaloff>:/conf_app/conf_app.ico</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <widget class="QToolBox" name="toolBox">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="page_3">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>496</width>
         <height>251</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Connection Properties</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="lbHost">
          <property name="text">
           <string>Host</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>lineHost</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineHost"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="lbDataSource">
          <property name="text">
           <string>Database</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>lineDatabase</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="lineDatabase"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="lbUsername">
          <property name="text">
           <string>Username</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
          <property name="buddy">
           <cstring>lineUsername</cstring>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="lineUsername"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="lbPassword">
          <property name="text">
           <string>Password</string>
          </property>
          <property name="buddy">
           <cstring>linePassword</cstring>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="linePassword">
          <property name="echoMode">
           <enum>QLineEdit::Password</enum>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="lbDriver">
          <property name="text">
           <string>Driver</string>
          </property>
          <property name="buddy">
           <cstring>cmbDriver</cstring>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="cmbDriver"/>
        </item>
        <item row="5" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="pushConnect">
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="conf_app.qrc">
              <normaloff>:/conf_app/encrypted.png</normaloff>:/conf_app/encrypted.png</iconset>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushDisconnect">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="conf_app.qrc">
              <normaloff>:/conf_app/decrypted.png</normaloff>:/conf_app/decrypted.png</iconset>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_4">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>161</width>
         <height>121</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Global Settings</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupSettings">
          <property name="title">
           <string/>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Country</string>
               </property>
               <property name="buddy">
                <cstring>cmbCountry</cstring>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="cmbCountry"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>City</string>
               </property>
               <property name="buddy">
                <cstring>cmbCity</cstring>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="cmbCity"/>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="pushApply">
               <property name="text">
                <string>Apply</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>190</width>
         <height>162</height>
        </rect>
       </property>
       <attribute name="label">
        <string>Edit Tables</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupTables">
          <property name="toolTip">
           <string>This control allows the edition of any database table, including inserting and removing records.
 Choose a table listed in the combo box, edit it by changing values and using the bottoms on the left to add/remove rows, and submit changes by pressing apply.
 To enable edition again, you just have to select another table.
 *Attention*: this function operates directly on the database, so there are no undos - be carefull with any changes you make! </string>
          </property>
          <property name="statusTip">
           <string>This control allows the edition of any database table, including inserting and removing records. </string>
          </property>
          <property name="whatsThis">
           <string>This control allows the edition of any database table, including inserting and removing records.
 Choose a table listed in the combo box, edit it by changing values and using the bottoms on the left to add/remove rows, and submit changes by pressing apply.
 To enable edition again, you just have to select another table.
 *Attention*: this function operates directly on the database, so there are no undos - be carefull with any changes you make! </string>
          </property>
          <property name="title">
           <string/>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Choose Table:</string>
               </property>
               <property name="buddy">
                <cstring>cmbTables</cstring>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="cmbTables">
               <property name="toolTip">
                <string>Database tables</string>
               </property>
               <property name="statusTip">
                <string>Database tables</string>
               </property>
               <property name="whatsThis">
                <string>This control lists database tables.</string>
               </property>
               <property name="editable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_2">
               <item>
                <widget class="QPushButton" name="pushInsert">
                 <property name="toolTip">
                  <string>Insert row</string>
                 </property>
                 <property name="statusTip">
                  <string>Insert row</string>
                 </property>
                 <property name="whatsThis">
                  <string>Inserts row into table</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="icon">
                  <iconset resource="conf_app.qrc">
                   <normaloff>:/conf_app/add.png</normaloff>:/conf_app/add.png</iconset>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="pushRemove">
                 <property name="toolTip">
                  <string>Remove row</string>
                 </property>
                 <property name="statusTip">
                  <string>Remove row</string>
                 </property>
                 <property name="whatsThis">
                  <string>Removes row into table</string>
                 </property>
                 <property name="text">
                  <string/>
                 </property>
                 <property name="icon">
                  <iconset resource="conf_app.qrc">
                   <normaloff>:/conf_app/cancel.png</normaloff>:/conf_app/cancel.png</iconset>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QTableView" name="tableView">
               <property name="alternatingRowColors">
                <bool>true</bool>
               </property>
               <property name="selectionMode">
                <enum>QAbstractItemView::SingleSelection</enum>
               </property>
               <property name="selectionBehavior">
                <enum>QAbstractItemView::SelectRows</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="pushApply_2">
               <property name="toolTip">
                <string>Apply changes</string>
               </property>
               <property name="statusTip">
                <string>Apply changes</string>
               </property>
               <property name="whatsThis">
                <string>Applies changes</string>
               </property>
               <property name="text">
                <string>Apply</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>514</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <widget class="QMenu" name="menuOptions">
     <property name="title">
      <string>Options</string>
     </property>
     <addaction name="actionShow_startup_message"/>
     <addaction name="actionShow_SQL_message"/>
    </widget>
    <widget class="QMenu" name="menuBackups">
     <property name="title">
      <string>Backups</string>
     </property>
     <addaction name="actionCreate_backup"/>
     <addaction name="actionRestore_backup"/>
    </widget>
    <addaction name="menuOptions"/>
    <addaction name="menuBackups"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
  </widget>
  <widget class="QToolBar" name="toolbar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="m_statusBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>18</height>
    </size>
   </property>
  </widget>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="conf_app.qrc">
     <normaloff>:/conf_app/exit.png</normaloff>:/conf_app/exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionShow_startup_message">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Show startup message</string>
   </property>
  </action>
  <action name="actionCreate_backup">
   <property name="icon">
    <iconset resource="conf_app.qrc">
     <normaloff>:/conf_app/folder_outbox.png</normaloff>:/conf_app/folder_outbox.png</iconset>
   </property>
   <property name="text">
    <string>Create backup</string>
   </property>
   <property name="statusTip">
    <string>Create backup from file</string>
   </property>
   <property name="whatsThis">
    <string>Creates backup from file</string>
   </property>
  </action>
  <action name="actionRestore_backup">
   <property name="icon">
    <iconset resource="conf_app.qrc">
     <normaloff>:/conf_app/folder_inbox.png</normaloff>:/conf_app/folder_inbox.png</iconset>
   </property>
   <property name="text">
    <string>Restore backup</string>
   </property>
   <property name="statusTip">
    <string>Restore backup from file</string>
   </property>
   <property name="whatsThis">
    <string>Restores backup from file</string>
   </property>
  </action>
  <action name="actionShow_SQL_message">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show SQL messages</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>lineHost</tabstop>
  <tabstop>lineDatabase</tabstop>
  <tabstop>lineUsername</tabstop>
  <tabstop>linePassword</tabstop>
  <tabstop>cmbDriver</tabstop>
  <tabstop>pushConnect</tabstop>
  <tabstop>pushDisconnect</tabstop>
  <tabstop>cmbCountry</tabstop>
  <tabstop>cmbCity</tabstop>
  <tabstop>pushApply</tabstop>
  <tabstop>cmbTables</tabstop>
  <tabstop>tableView</tabstop>
  <tabstop>pushInsert</tabstop>
  <tabstop>pushRemove</tabstop>
  <tabstop>pushApply_2</tabstop>
 </tabstops>
 <resources>
  <include location="conf_app.qrc"/>
  <include location="../app_new/app_new.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>conf_appClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushConnect</sender>
   <signal>clicked()</signal>
   <receiver>conf_appClass</receiver>
   <slot>connectDB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>289</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushDisconnect</sender>
   <signal>clicked()</signal>
   <receiver>conf_appClass</receiver>
   <slot>disconnectDB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>393</x>
     <y>284</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbCountry</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>conf_appClass</receiver>
   <slot>filterModel(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushApply</sender>
   <signal>clicked()</signal>
   <receiver>conf_appClass</receiver>
   <slot>apply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>446</x>
     <y>315</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbCountry</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>conf_appClass</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbCity</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>conf_appClass</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>233</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cmbTables</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>conf_appClass</receiver>
   <slot>showTable(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushInsert</sender>
   <signal>clicked()</signal>
   <receiver>conf_appClass</receiver>
   <slot>insertRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>64</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushRemove</sender>
   <signal>clicked()</signal>
   <receiver>conf_appClass</receiver>
   <slot>removeRow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>64</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushApply_2</sender>
   <signal>clicked()</signal>
   <receiver>conf_appClass</receiver>
   <slot>applyChanges2Table()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>67</x>
     <y>64</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_SQL_message</sender>
   <signal>triggered(bool)</signal>
   <receiver>conf_appClass</receiver>
   <slot>showSqlMessages(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>connectDB()</slot>
  <slot>disconnectDB()</slot>
  <slot>filterModel(QString)</slot>
  <slot>apply()</slot>
  <slot>enableApply()</slot>
  <slot>showTable(QString)</slot>
  <slot>insertRow()</slot>
  <slot>removeRow()</slot>
  <slot>applyChanges2Table()</slot>
  <slot>showSqlMessages(bool)</slot>
 </slots>
</ui>
